### ISSUE TITLE
When opening a post, show all replies (not just those from servers known to the user's instance).

### ISSUE THREAD

**Issue by larryedwards** [mastodon/mastodon#22674]
### Pitch

If the following is true, I hope a feature can be added to make the display of replies complete (OP link: https://fedi.simonwillison.net/@simon/109559268498004036) 

"simon@simonwillison.net  Simon Willison

"My least favourite sharp edge of Mastodon is the fact that when you view someone else's post you only see replies to it that are known to your server - so there's actually a good chance there will be relies that are completely invisible to you, especially if you run your own instance

"I'd love it if tapping a post kicked off a request back to the original server that fetched the current reply count, and provided a "view all replies" button if there were replies not yet visible to me

### Motivation

Full access the the discussion of a topic is highly important for some users (e.g. those on Mastodon regarding issues on climate science, energy matters, or policy development, etc.).  Also, it is important for finding new people to follow or for others to find you who otherwise wouldn't.

**Comment by DeadWisdom** [comment]
This might be a bug in the implementation of Activity Pub:

https://www.w3.org/TR/activitypub/#inbox-forwarding

**Comment by simonw** [comment]
I don't think this is the same problem as inbox forwarding - which Mastodon already implements.

Inbox forwarding works to ensure that everyone who follows a specific user gets sent replies to that user's post. Mastodon implements this - I've seen it in action, watching as my Sidekiq queue gets /really/ busy forwarding replies when I start a noisy conversation which attracts a lot of comments.

The issue here is different: it happens when I view a conversation on a post by someone who I don't already follow - eg because their original post was boosted into my timeline.

It's also primarily an issue that affects people like me who run their own instances (or are members of a tiny instance). If you're using https://mastodon.social/ you'll rarely see a problem like this as much more of the overall Mastodon content flow is already being cached by your home server.

**Comment by jai-o** [comment]
+1

For clarity:

Scenario A:
If you're **not logged in**, with the "Copy the post URL" or "Open original page", you will then see all replies, except, if the reply comes from a moderated server (suspended/blocked). Correct? Or, do you really see all replies?

Scenario B:
If you're **logged in**, then what replies are shown need to also account for if you have blocked any user or any domain, plus if your instance has blocked another server. This assumes the reply was created before the blocking. Therefore, another logged-in user who did not block any users/domains, and their instance did not block another server, this user would see all the replies. Correct?

**Comment by rimu** [comment]
Fundamentally, this issue is the same as https://github.com/mastodon/mastodon/issues/34 (which was created **6 years** ago). To show replies means retrieving them from the other server then inserting them into the local instances' database which will mean they show up in the Federated timeline as if they were freshly posted toots.

**Comment by brendanjones** [comment]
Isn't this 100% a duplicate of https://github.com/mastodon/mastodon/issues/9409 ?

**Comment by mdemoss** [comment]
> everyone who follows a specific user gets sent replies to that user's post

I see that this very often isn't the case on my single-user instance. v4.2.10.
This includes accounts I follow on mastodon.social.

**Comment by simonw** [comment]
Looks like this just got fixed!

 - #9409

That improvement shipped in Mastodon 4.5: https://blog.joinmastodon.org/2025/11/mastodon-4.5/
